Programs/_testembed.c

```diff
@@ -1296,6 +1319,49 @@ static int test_audit(void)
     return result;
 }
 
+static int test_audit_tuple(void)
+{
+    Py_ssize_t sawSet = 0;
+
+    // we need at least one hook, otherwise code checking for
+    // PySys_AuditTuple() is skipped.
+    PySys_AddAuditHook(_audit_hook, &sawSet);
+    _testembed_Py_InitializeFromConfig();
+
+    assert(!PyErr_Occurred());
+
+    // pass Python tuple object
+    PyObject *tuple = Py_BuildValue("(i)", 444);
+    if (tuple == NULL) {
+        goto error;
+    }
+    assert(PySys_AuditTuple("_testembed.set", tuple) == 0);
```

Comment:
I dislike using `assert` in native tests, as it can abort the rest of the test suite and/or display modal UI to the user.

I'd prefer the checks to return an error code. It can be wrapped up in a macro if that makes life easier (I'm not concerned about ref leaks in test failures). The macro can use `__FILE__` and `__LINE__` to identify the exact source of the error, but should `return` rather than aborting.

Reply:
Ok, done